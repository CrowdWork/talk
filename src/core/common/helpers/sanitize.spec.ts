import { JSDOM } from "jsdom";

import { createSanitize } from "./sanitize";

const window = new JSDOM("", {}).window;
const sanitize = createSanitize(window as any);

it("sanitizes out tags not allowed", () => {
  expect(
    sanitize("<script type=\"text/javascript\">alert('ok');</script>")
  ).toMatchSnapshot();

  expect(sanitize("<script src=malicious-code.js></script>")).toMatchSnapshot();
});

it("sanitizes out attributes not allowed", () => {
  expect(sanitize('<div id="test">Test</div>')).toMatchSnapshot();
});

it("allows anchor links", () => {
  expect(
    sanitize('<a href="http://test.com">This is a link</a>')
  ).toMatchSnapshot();
});

it("allows mailto links", () => {
  expect(
    sanitize('<a href="mailto:email@example.com">email@example.com</a>')
  ).toMatchSnapshot();
});

it("allows anchor tags and counts them correctly", () => {
  const el = sanitize(
    `
    <div>
      <a href="https://mozilla.org/">Mozilla</a>
      <a href="https://mozilla.org/">Mozilla</a>
    </div>
  `
  );

  expect(el.innerHTML).toMatchSnapshot();
  expect(el.getElementsByTagName("a").length).toEqual(2);
});

it("allows bolded tags", () => {
  expect(sanitize("A <b>bolded comment!</b>")).toMatchSnapshot();
});
