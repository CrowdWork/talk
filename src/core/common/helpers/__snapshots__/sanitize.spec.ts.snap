// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows anchor links 1`] = `
<body>
  <a
    href="http://test.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    http://test.com
  </a>
</body>
`;

exports[`allows anchor tags and counts them correctly 1`] = `
"
    <div>
      <a href=\\"https://mozilla.org/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">https://mozilla.org/</a>
      <a href=\\"https://mozilla.org/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">https://mozilla.org/</a>
    </div>
  "
`;

exports[`allows bolded tags 1`] = `
<body>
  A 
  bolded comment!
</body>
`;

exports[`allows mailto links 1`] = `
<body>
  <a
    href="mailto:email@example.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    email@example.com
  </a>
</body>
`;

exports[`sanitizes out attributes not allowed 1`] = `
<body>
  <div>
    Test
  </div>
</body>
`;

exports[`sanitizes out tags not allowed 1`] = `<body />`;

exports[`sanitizes out tags not allowed 2`] = `<body />`;
